pkg/operator/ceph/object/json_helpers.go

```diff
@@ -89,6 +89,51 @@ func setObjProperty[T string | []string | map[string]interface{} | []interface{}
 	return prev, fmt.Errorf("json property %q not found", strings.Join(path, "."))
 }
 
+// setObjPropertyIfExists - helper function to manipulate JSON Objects.
+// sets new value to json object nested field only if it is already exists in json and returns previous value.
+func setObjPropertyIfExists[T string | []string | map[string]interface{} | []interface{}](obj map[string]interface{}, val T, path ...string) (T, error) {
```

Comment:
This function looks to be ~90% copy-pasted from `setObjProperty()`. Is there a good way to abstract out the main logic into a helper, that can be used from both `setObjProperty()` and `setObjPropertyIfExists()`? Or alternately, could `setObjPropertyIfExists()` be called by `setObjProperty()`? 

I think this will be best for long term maintenance. If there are any logic corner cases we need to add, it will help to be able to add/fix the logic in only one function.